src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java

```diff
@@ -272,7 +272,8 @@ public void deletePullRequestApproval(String pullRequestId) throws IOException,
      */
     @Override
     public boolean checkPathExists(@NonNull String branch, @NonNull String path) throws IOException, InterruptedException {
-        int status = getRequestStatus(V1_API_BASE_URL + owner + "/" + repositoryName + "/raw/" + branch + "/" + path);
+        int status = getRequestStatus(V1_API_BASE_URL + owner + "/" + repositoryName + "/raw/" +
+                branch.replaceAll(" ", "%20") + "/" + path);
```

Comment:
@alexbrjo There are other special characters that we should probably URL-encode.  

Recommend using https://google.github.io/guava/releases/19.0/api/docs/com/google/common/net/UrlEscapers.html or similar to escape URLEncode characters (note: the built-in URLEncoder class isn't safe for this because it's meant for form data and handles spaces differently). 

Unless we can prove that Mercurial doesn't allow other reserved URL characters in the branch names (Git does).